### DISCUSSION TITLE
Regular Expressions - Nachfragen zu Beispiel aus der Vorlesung

### DISCUSSION THREAD

**Discussion by MrsMaulwurf** [irgroup-classrooms/dis08-2021#37] [Q&A] [answered]
Hallo,
ich habe den Vorlesungsmitschnitt verfolgt, bin mir aber mit meinen eigenen abweichenden Lösungen unsicher. Wären folgende Lösungen auch korrekt (oder schließen sich manche Kombinationen von Patterns aus)?

Folie 26
\w [^xyz]

Folie 46
^ (file\w+)\.pdf$

Vielen Dank für die Hilfe.

Viele Grüße
Tina Grahl

**Comment by HaakFab** [comment, marked as answer]
Hallo,
Die beiden Lösungen würden so nicht ganz funktionieren (kurzer Edit):
\w [^xyz]
> "[ ]" beschreibt eine Menge, z. B. würde [abc] a ODER b ODER c matchen, aber immer nur ein Zeichen
> \w matcht ohne weiteren operator (., *, + oder ?) auch nur ein Zeichen
daher würden hier nur 2 Zeichen gematched

^(file\w+).pdf$

> hier ist \w das problem, da nur alphanummerische Zeichen, aber nicht z. B. _ gematcht wird (kommt aber in Dateinamen vor)

Ich hoffe, die Erklärung hat Ihnen geholfen!

VG,
Fabian Haak

**Comment by HaakFab** [comment]
kurzer Nachtrag:
auf https://regex101.com/ können Sie super testen! Wenn man mit der Maus über eine eingegebene RegEx fährt, bekommt man auch Erklärungen zu allen verwendeten Zeichen.
VG

**Comment by HaakFab** [comment]
Und noch ein Nachtrag (weil ich mir nicht sicher bin, ob das unbewusst oder für Lesbarkeit gemacht wurde):
Leerzeichen sind auch Zeichen! also die RegEx (ab ab) matcht abab nicht!

**Reply by MrsMaulwurf** [reply]
Nein, ist auch noch ein Fehler :(

Danke für die vielen Antworten!
